src/compiler/checker.ts

```diff
@@ -12650,6 +12681,37 @@ namespace ts {
             }
         }
 
+        function getLeftmostAliasDeclarationFromExpressionOrEnityName(node: Expression | EntityName): Declaration {
+            switch (node.kind) {
+                case SyntaxKind.PropertyAccessExpression:
+                    let result = getLeftmostAliasDeclarationFromExpressionOrEnityName((<PropertyAccessExpression>node).expression);
+                    if (result) {
+                        return result;
+                    }
+                    break;
+
+                case SyntaxKind.QualifiedName:
+                    let result2 = getLeftmostAliasDeclarationFromExpressionOrEnityName((<QualifiedName>node).left);
+                    if (result2) {
+                        return result2;
+                    }
+
+                case SyntaxKind.Identifier:
+                    break;
+
+                default:
+                    return;
+            }
+
+            let meaning = node.parent.kind === SyntaxKind.QualifiedName || node.parent.kind === SyntaxKind.PropertyAccessExpression?
+                SymbolFlags.Namespace :
+                SymbolFlags.Value | SymbolFlags.Type | SymbolFlags.Namespace;
+            let resolvedSymbol = resolveEntityName(node, meaning | SymbolFlags.Alias, /*ignoreErrors*/ true);
+            if (resolvedSymbol && !!(resolvedSymbol.flags & SymbolFlags.Alias)) {

```

Comment:
It is because: eg.
```TypeScript
    module M {
        export module Q {
            export class C {
            }
        }
    
        export class C {
        }
        export import im = Q.C;
    }
    class B1 extends im1.C { // this should be reported as error
    }
    import im1 = M;

```